odoo/addons/base/ir/ir_mail_server.py

```diff
@@ -464,6 +470,47 @@ def send_email(self, message, mail_server_id=None, smtp_server=None, smtp_port=N
             raise MailDeliveryException(_("Mail Delivery Failed"), msg)
         return message_id
 
+    @api.multi
+    def _update_message_from(self, message):
+        """Update the FROM header on message to align with server settings.
+
+        Args:
+            message (email.message): The message object to be evaluated and
+                possibly mutated.
+        """
+
+        self.ensure_one()
+
+        if not self.smtp_from:
+            return
+
+        message.replace_header('Reply-To', message['From'])
+        message.replace_header(
+            'From', self.__get_canonical_from(message['From']),
+        )
+
+    @api.multi
+    def __get_canonical_from(self, original_from):
```

Comment:
Why use a mangled method name? This generally complicates access with little actual actual benefit.

Reply:
Just to confirm, by mangled you mean a double underscore prefix?

Reply:
Yes, [exactly](https://docs.python.org/2/tutorial/classes.html#private-variables-and-class-local-references) :)

Reply:
Sweet good point - in hindsight it's actually counter-intuitive to why I made this a separate method anyways. The goal was to allow addons to modify this behavior, but good luck with a private!